src/py/flwr/server/superlink/fleet/grpc_rere/server_interceptor.py

```diff
@@ -124,45 +125,35 @@ def _generic_method_handler(
                     _PUBLIC_KEY_HEADER, context.invocation_metadata()
                 )
             )
-            is_public_key_known = client_public_key_bytes in self.client_public_keys
-            if not is_public_key_known:
+            if client_public_key_bytes not in self.client_public_keys:
                 context.abort(grpc.StatusCode.UNAUTHENTICATED, "Access denied")
 
             if isinstance(request, CreateNodeRequest):
-                context.send_initial_metadata(
-                    (
-                        (
-                            _PUBLIC_KEY_HEADER,
-                            self.encoded_server_public_key,
-                        ),
-                    )
+                return self._create_authenticated_node(
+                    client_public_key_bytes, request, context, method_handler
                 )
-            elif isinstance(
-                request,
-                (
-                    DeleteNodeRequest,
-                    PullTaskInsRequest,
-                    PushTaskResRequest,
-                    GetRunRequest,
-                    PingRequest,
-                ),
-            ):
-                hmac_value = base64.urlsafe_b64decode(
-                    _get_value_from_tuples(
-                        _AUTH_TOKEN_HEADER, context.invocation_metadata()
-                    )
-                )
-                client_public_key = bytes_to_public_key(client_public_key_bytes)
-                shared_secret = generate_shared_key(
-                    self.server_private_key,
-                    client_public_key,
-                )
-                verify = verify_hmac(
-                    shared_secret, request.SerializeToString(True), hmac_value
+
+            if isinstance(request, PingRequest):
+                print("Ping, ", request)
```

Comment:
Never use `print` in framework code (always use `log`)

Reply:
Sorry, I include this to check the CI, because the e2e is still failing. In my machine there is a Ping request and no fail.

Reply:
I will remove after this.

Reply:
Removed.